core/Archive/ArchiveInvalidator.php

```diff
@@ -230,6 +230,15 @@ private function getPeriodsToInvalidate($dates, $periodType, $cascadeDown)
     {
         $periodsToInvalidate = array();
 
+        // Get the individual dates for a range, we need to check all of them
+        if ($periodType === 'range') {
+            $allDates = array();
+            for ($date = $dates[0]; $date->getTimestamp() <= $dates[1]->getTimestamp(); $date = $date->addDay(1)) {
+                $allDates[] = $date;
+            }
+            $dates = $allDates;
+        }
```

Comment:
If I'm getting this right (and I probably am not, the invalidation code always confuses me), the reason we check for all subperiods is because ranges can span multiple months? In this case could we do `addMonth(1)` or something to check less dates? I think this might result in slightly more readable SQL, though that may not be too important in the long run.

Reply:
It's for catching smaller subperiods that lie within the month. If we just check one day each month we'll miss the ones that don't include that day.

Reply:
Again, I could be seeing it incorrectly, but I was thinking of using date pairs like `2019-01-01,2019-01-31` so we'd check if there's any range affecting that whole month, rather than individual days.

Reply:
I discussed this yesterday with @mattab and we have decided that although it would make the SQL easier to read and possibly be more efficient too, it's not worth investing the time as we don't expect that the query will be run very often.

Reply:
No worries.